libs/s25main/network/GameClient.h

```diff
@@ -80,7 +81,8 @@ class GameClient final :
     bool Connect(const std::string& server, const std::string& password, ServerType servertyp, unsigned short port,
                  bool host, bool use_ipv6);
     /// Start the server and connect to it
-    bool HostGame(const CreateServerInfo& csi, const boost::filesystem::path& map_path, MapType map_type);
+    bool HostGame(const CreateServerInfo& csi, const boost::filesystem::path& map_path,
```

Comment:
somehow having so many parameters in the mean time it should probably be best to use a struct for it:
```suggestion
struct MapInfo
{
boost::filesystem::path map_path;
MapType map_type;
boost::optional<boost::filesystem::path lua_path;
};
bool HostGame(const CreateServerInfo& csi, const MapInfo& map);
```

so we don't need to use `boost::none` but can use struct initialization.

Reply:
Use name MapDescription because MapInfo already exists for other parameters.